src/append_only_data.rs

```diff
@@ -829,6 +829,15 @@ impl Data {
         }
     }
 
+    pub fn get(&self, key: Vec<u8>) -> Option<&Vec<u8>> {
```

Comment:
This should be
```suggestion
    pub fn get(&self, key: &[u8]) -> Option<&Vec<u8>> {
```
and we wouldn't need to call `.as_slice()` for `key` then